clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1438,7 +1438,8 @@ def FuncOp : CIR_Op<"func", [
                        OptionalAttr<DictArrayAttr>:$arg_attrs,
                        OptionalAttr<DictArrayAttr>:$res_attrs,
                        OptionalAttr<FlatSymbolRefAttr>:$aliasee,
-                       OptionalAttr<ASTFunctionDeclAttr>:$ast);
+                       OptionalAttr<ASTFunctionDeclAttr>:$ast,
+                       OptionalAttr<NoProtoFuncDeclAttr>:$no_proto);
```

Comment:
Thanks for adding the attribute. I believe this could be a bit more simple though, a `UnitAttr:$no_proto` does the job!